### PR TITLE
adding clusterResource analyzer docs

### PR DESCRIPTION
**Pull request by CpuID**
ref https://github.com/replicatedhq/troubleshoot/pull/780 and https://github.com/replicatedhq/troubleshoot/pull/1210

I'm going to digest my thoughts here before marking it "ready", but so far I think it's more than we have right now...

I may augment it further, suggestions welcome for specific examples also.

### CODE REVIEW

**Comment by CpuID** [comment]
this also closes https://github.com/replicatedhq/troubleshoot/issues/1161

File: docs/source/analyze/cluster-resource.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+Note: when using arithmetic comparison in `when`, this is not supported at this time.
+
+## Outcomes
+
+The `when` value in an outcome of this analyzer can accept a few variations.
+
+If the `yamlPath` points to a quantity-based value (eg. the size or quota or something), standard arithmetic comparison operators may be used.
+eg. `<`, `<=`, `>`, `>=`, `==`, `!=`. 
+
+Alternatively if `regex` specifies an expected value, using a boolean `true` or `false` in the `when` clause is acceptable.
```
Comment (DexterYan) [inline docs/source/analyze/cluster-resource.md:44]: in the yaml, we are using the string true and false, should we need to clarify here or support both?
```
- pass:
     when: "true"  or  true ?
     message: is ReadWriteOnce
```

Comment (CpuID) [reply docs/source/analyze/cluster-resource.md]: fixed in https://github.com/replicatedhq/troubleshoot.sh/pull/503/commits/e80d0345093c8bf0732a6785d100fd2b10ffaf08

File: docs/source/analyze/cluster-resource.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+## Parameters
+
+**checkName**: (Optional) Analyzer name.
+Used for uniqueness if multiple analyzers are defined with similar parameters.
+
+**kind**: (Required) The type of Kubernetes resource being targeted by `name`.
+
+**name**: (Required) The name of the resource to check.
+
+**namespace**: (Optional) The namespace to look for the resource in.
```
Comment (danj-replicated) [inline docs/source/analyze/cluster-resource.md:27]: it's worth noting here that the absence of `namespace` configures the analyzer to search for cluster-scoped resources

**Review by danj-replicated — DISMISSED** [review]

File: docs/source/analyze/cluster-resource.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+**Note:** This is not supported when using arithmetic comparison in `when`.
+
+**regexGroups**: (Optional) See [Regular Expression](/analyze/regex) documentation.
+**Note:** This is not supported when using arithmetic comparison in `when`.
+
+## Outcomes
+
+The `when` value in an outcome of this analyzer can accept a few variations.
+
+If the `yamlPath` points to a quantity-based value (such as the size or quota or something), standard arithmetic comparison operators can be used: `<`, `<=`, `>`, `>=`, `==`, `!=`.
```
Comment (jonquil2002) [inline docs/source/analyze/cluster-resource.md:41]: What is meant by "or something" here? Is it a value or can it be removed?

Comment (CpuID) [reply docs/source/analyze/cluster-resource.md:41]: can be removed, not a specific value

Comment (CpuID) [reply docs/source/analyze/cluster-resource.md]: sorted, I botched a merge commit but reverted it with a force push and fixed it again.

removed the `or something`

**Review by DexterYan — DISMISSED** [review]

File: docs/source/analyze/cluster-resource.md (hunk 2 above)
Comment (CpuID) [reply docs/source/analyze/cluster-resource.md]: done thx

**Review by jonquil2002 — APPROVED** [review]
lgtm

**Review by xavpaice — CHANGES_REQUESTED** [review]
note: this needs an additional entry in https://github.com/replicatedhq/troubleshoot.sh/blob/main/docs/gatsby-config.js also - so it gets to show up in the index

**Comment by CpuID** [comment]
> note: this needs an additional entry in https://github.com/replicatedhq/troubleshoot.sh/blob/main/docs/gatsby-config.js also - so it gets to show up in the index

good catch thx, fixed in https://github.com/replicatedhq/troubleshoot.sh/pull/503/commits/351d545cf03e85e9a79384efb36924cd051ea375

**Review by xavpaice — APPROVED** [review]

**Pull request merged**